File: angular/tour-of-heroes/cat-manager/src/app/cat.ts
export class Cat {
    static idCounter: number = 0;

    id: number;
    name: string;
    fur: string;

    constructor(name: string, fur: string) {
        this.name = name;
        this.fur = fur;
        this.id = Cat.idCounter++;
    }
}

File: angular/tour-of-heroes/C#-back-end/CatManagerBackEnd/Models/Cat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CatManagerBackEnd.Models
{
    public class Cat
    {
        public int id;
        public string name;
        public string fur;

        public Cat(int id, string name, string fur) {
            this.id = id;
            this.name = name;
            this.fur = fur;
        }
    }
}
